datasource/src/test/scala/quasar/physical/blobstore/BlobstoreDatasourceSpec.scala

```diff
@@ -71,6 +77,16 @@ abstract class BlobstoreDatasourceSpec[F[_]: Effect] extends EffectfulQSpec[F] {
           ResourceName("testdata") -> ResourcePathType.prefix))
     }
 
+    "list children at the root of the bucket (alternative)" >>* {
+      assertPrefixedChildPaths(
+        ResourcePath.root() / ResourceName(""),
```

Comment:
UI generates calls with a trailing slash like this:`curl http://localhost:8080/datasource/82c6237c-6a82-4b04-b710-74b58bb94f56/discovery/` (which seems ok), which then gets translated to `ResourcePath.root() / ResourceName("")`. This is understandable (ie. to distinguish it from an url that doesn't end with slash) but also confusing. 

I choose to resolve this in azure because other datasources apparently already handle this correctly, but there may be a better (but also more impactful) way to resolve this centrally in quasar.

Reply:
Yeah, this should probably be resolved in sdbe so that the trailing `ResourceName("")` is not produced.

Reply:
Can confirm we have special handling for this in `quasar-s3`.